hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -848,6 +1011,21 @@ ClientConnectionManagerImpl::connect_to_all_cluster_members()
 
     for (const auto& member :
          client_.get_client_cluster_service().get_member_list()) {
+
+        {
+            std::lock_guard<std::recursive_mutex> guard{ client_state_mutex_ };
+
+            if (client_state_ == client_state::SWITCHING_CLUSTER ||
```

Comment:
This seems to be in wrong place. It should be in method `ClientConnectionManagerImpl::connect_to_all_members`.

Reply:
Fixed.